### ISSUE TITLE
Component package for Future may need dependency on Deferred

### ISSUE THREAD

**Issue by StuffOfInterest** [zzzprojects/EntityFramework-Plus#9]
I was playing around with the future queries support in EF+ this morning.  Being that I wasn't interested in the other features of EF+ I only pulled down the NuGet package for future queries.  In testing that I found that I needed the deferred queries package to support value results (such as Count()).  With both packages installed I found that it wasn't possible to chain FutureValue() onto DeferredCount().  After that I uninstalled both packages and installed the full EF+ package.  With that, it was possible to chain the method calls.

Being that there are pieces of Future which depend on Deferred, maybe it would make sense to have the Future package dependent on Deferred so that the full capabilities of Future can be utilized.

**Comment by zzzprojects** [comment]
It should already be possible to use both methods using standalone version.

The following code work has been tested using NuGet standalone version (1.0.4).

```
using (var ctx = new TestContext())
{
    ctx.Entity_Basics.DeferredCount().FutureValue();
}
```

Can you show me the example which was not working?

**Comment by StuffOfInterest** [comment]
OK, that is odd.  Must have been an issue with Visual Studio or ReSharper.  When I did this in the morning I installed the QueryFuture package first and later the QueryDeferred package.  VS (or maybe ReSharper) would not resolve the linkage.  I just went back and installed QueryDeferred and then QueryFuture before typing any code.  This time it resolved just fine.

I'm going back to fighting with MiniProfiler not displaying batched queries, which is not the fault of your work but is an issue with MiniProfiler on EF6.

Thanks.
